### PR TITLE
Rwa 1325 priority

### PR DESCRIPTION
**Pull request by chrisdavecm**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-1325

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/request/enums/TaskAttributeDefinition.java [hunk 1]
```diff
@@ -43,7 +43,8 @@ public enum TaskAttributeDefinition {
     TASK_TITLE("task_title", new TypeReference<String>() {}),
     TASK_TYPE("task_type", new TypeReference<String>() {}),
     TASK_WORK_TYPE("task_work_type", new TypeReference<String>() {}),
-    TASK_ADDITIONAL_PROPERTIES("task_additional_properties", new TypeReference<Map<String, String>>() {});
+    TASK_ADDITIONAL_PROPERTIES("task_additional_properties", new TypeReference<Map<String, String>>() {}),
+    TASK_PRIORITY_DATE("task_priority_date", new TypeReference<Map<String, String>>() {});
```
Comment (dharmendrak) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/request/enums/TaskAttributeDefinition.java:47]: this should be String or Date

**Review by dharmendrak — DISMISSED** [review]
one small comment, please action it

**Review by dharmendrak — APPROVED** [review]

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java [hunk 2]
```diff
@@ -274,6 +279,82 @@ public void should_return_a_201_when_initiating_a_specific_access_task_by_id() {
...
+                                             + "trigger/decideAnApplication)"))
+            .body("task_role_resources.size()", equalTo(10))
+            .body("additional_properties", equalToObject(Map.of(
+                "key1", "value1",
+                "key2", "value2",
+                "key3", "value3",
+                "key4", "value4"
+            ))).body("minor_priority", equalTo(500))
+            .body("major_priority", equalTo(1000))
+            .body("priority_date", notNullValue());
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java:353]: as well as notNullValue, should we also assert that it is set to the taskAttributes.dueDate?

File: src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java [hunk 3]
```diff
@@ -274,6 +280,86 @@ public void should_return_a_201_when_initiating_a_specific_access_task_by_id() {
...
+            .body("description", equalTo("[Decide an application](/case/WA/WaCaseType/${[CASE_REFERENCE]}/"
+                                             + "trigger/decideAnApplication)"))
+            .body("task_role_resources.size()", equalTo(10))
+            .body("additional_properties", equalToObject(Map.of(
+                "key1", "value1",
+                "key2", "value2",
+                "key3", "value3",
+                "key4", "value4"
+            ))).body("minor_priority", equalTo(500))
+            .body("major_priority", equalTo(5000));
```
Comment (BenNzewi) [inline src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskInitiateByIdControllerTest.java:358]: can we add the priority_date = dueDate assertion?

**Review by rasikasri — DISMISSED** [review]

**Review by BenNzewi — COMMENTED** [review]
Can we also add assertions to our search Task endpoint(s) FT the returns tasks to verify these 3 attributes will now be returned?

One 200 test should be ok per FT.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java [hunk 4]
```diff
@@ -105,8 +107,12 @@ public TaskResource mapToTaskResource(String taskId, List<TaskAttribute> taskAtt
         ExecutionTypeResource executionTypeResource = extractExecutionType(attributes);
         OffsetDateTime dueDate = readDate(attributes, TASK_DUE_DATE, null);
         OffsetDateTime createdDate = readDate(attributes, TASK_CREATED, ZonedDateTime.now().toOffsetDateTime());
+        OffsetDateTime priorityDate = readDate(attributes, TASK_PRIORITY_DATE, null);
 
         Objects.requireNonNull(dueDate, "TASK_DUE_DATE must not be null");
+        if (priorityDate == null) {
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java:113]: unit test coverage for new line of logic.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java [hunk 5]
```diff
@@ -190,6 +197,9 @@ public Task mapToTaskWithPermissions(TaskResource taskResource, Set<PermissionTy
             taskResource.getRoleCategory(),
             taskResource.getDescription(),
             taskResource.getAdditionalProperties(),
+            taskResource.getMinorPriority(),
+            taskResource.getMajorPriority(),
+            taskResource.getPriorityDate() == null ? null : taskResource.getPriorityDate().toZonedDateTime(),
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java:202]: unit test coverage for new line of logic.

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java [hunk 6]
```diff
@@ -442,6 +452,15 @@ private void mapVariableToTaskResourceProperty(TaskResource taskResource, String
                     Map<String, String> additionalProperties = extractAdditionalProperties(value);
                     taskResource.setAdditionalProperties(additionalProperties);
                     break;
+                case MINOR_PRIORITY:
```
Comment (BenNzewi) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java:455]: Lines 456 - 463 unit test?

**Review by dharmendrak — APPROVED** [review]

**Pull request merged**